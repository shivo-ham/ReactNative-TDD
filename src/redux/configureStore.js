import { createStore } from "redux";
import { applyMiddleware } from "redux";
import thunk from "redux-thunk";
import rootReducer from "./reducers/rootReducer";
const configureStore = () => {
  return createStore(rootReducer, applyMiddleware(thunk));
};
export default configureStore;
