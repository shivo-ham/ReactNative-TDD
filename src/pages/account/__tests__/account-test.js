/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
/**
 * @format
 */

import { act, waitFor } from "@testing-library/react-native";
import React from "react";
import "react-native";
import { FlatList, Text } from "react-native";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import Account from "../";
import configureStore from "../../../redux/configureStore";
jest.mock("@react-navigation/native", () => {
  return {
    useFocusEffect: (callback) => callback(),
    useIsFocused: () => true
  };
});

describe("Account", () => {
  const fakeNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn()
  };

  const route = {
    params: {
      token: "token"
    }
  };
  const store = configureStore();
  it("should render correctly", () => {
    let rendered;
    act(() => {
      renderer
        .create(
          <Provider store={store}>
            <Account navigation={fakeNavigation} route={route} />
          </Provider>
        )
        .toJSON();
    });
    waitFor(() => {
      expect(rendered).toMatchSnapshot();
    });
  });

  jest.mock("@react-native-firebase/database", () => {
    const set = jest.fn();
    return {
      database: jest.fn(() => ({
        ref: jest.fn(() => ({
          once: jest.fn(() => ({
            then: jest.fn(() => ({
              set
            }))
          }))
        }))
      }))
    };
  });

  it("renders the FlatList component", () => {
    const tree = renderer
      .create(
        <FlatList
          data={["apple", "banana", "kiwi"]}
          keyExtractor={(item) => item}
          renderItem={({ item }) => <Text>{item}</Text>}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
