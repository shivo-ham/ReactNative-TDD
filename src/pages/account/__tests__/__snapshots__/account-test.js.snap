// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account renders the FlatList component 1`] = `
<RCTScrollView
  data={
    Array [
      "apple",
      "banana",
      "kiwi",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        apple
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        banana
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        kiwi
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Account should render correctly 1`] = `undefined`;
